### PR TITLE
Fix xtrigger `sequential` arg validation & improve docs

### PR DESCRIPTION
**Pull request by MetRonnie**
Built on/needs:
- #5809

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] No dependency changes 
- [x] Tests are included 
- [x] No changelog needed
- [x] Works with https://github.com/cylc/cylc-doc/pull/727
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

**Comment by oliver-sanders** [comment]
(LGTM)

File: cylc/flow/dbstatecheck.py [hunk 1]
```diff
@@ -84,8 +84,16 @@ def __init__(self, rund, workflow, db_path=None):
                 self.db_point_fmt = self._get_db_point_format_compat()
                 self.c7_back_compat_mode = True
             except sqlite3.OperationalError:
+                self.conn.close()
```
Comment (oliver-sanders) [inline cylc/flow/dbstatecheck.py:87]: Consider using `with suppress(Exception)`

**Review by wxtim — APPROVED** [review]
Looking good.

**Review by oliver-sanders — APPROVED** [review]

File: cylc/flow/dbstatecheck.py (hunk 1 above)
Comment (oliver-sanders) [reply cylc/flow/dbstatecheck.py:87]: (merge if you don't think this is relevant)

**Comment by MetRonnie** [comment]
Merging with Oliver's review suggestion force-pushed

**Pull request merged**